facade/logstash.go

```diff
@@ -215,7 +215,7 @@ func getFilters(configs []servicedefinition.LogConfig, filterDefs map[string]str
 		for _, filtName := range config.Filters {
 			//do not write duplicate types, logstash doesn't handle this
 			if !utils.StringInSlice(config.Type, *typeFilter) {
-				filters += fmt.Sprintf("\n  if [type] == \"%s\" {\n    %s\n  }\n",
+				filters += fmt.Sprintf("\n  if \"%s\" in [file] {\n    %s\n  }\n",
```

Comment:
Your solution assumes that the value of `config.Type` is always found in the file path defined by `[file]`, but that's not true in all cases. It's true sort of by accident for many case, but will not in the most general case. 
For instance, these lines will never produce a positive match:
```
if "zenossaudit" in [file] {
if "zope_eventlog" in [file] {
```
because the file for type=zenossaudit is named `/opt/zenoss/log/audit.log`, and the file for type=zenossaudit is named `/opt/zenoss/log/event.log`

I think you need to use something like `if [file] == "/opt/zenoss/log/zenmodeler.log"`, but you need to look for 2 things in the final output:
* there should be only one "if" stanza for "/opt/zenoss/log/audit.log" (this file appears in multiple services)
* there should be an "if" stanza for "/opt/zenoss/log/event.log"  and ""/opt/zenoss/log/Z2.log" (both of these log files are defined by the Zope service).